tracing/init_test.go

```diff
@@ -0,0 +1,43 @@
+package tracing
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestInit(t *testing.T) {
+	tt := []struct {
+		name        string
+		serviceName string
+
+		expectedErrorStr string
+	}{
+		{
+			name:        "success",
+			serviceName: "test",
+		},
+		{
+			name:        "failed to initialize jaeger tracer",
+			serviceName: "",
+
+			expectedErrorStr: "cannot initialize jaeger tracer",
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			tracer, closer, err := InitTracer("")
```

Comment:
isn't it better to just 

```
struct { 
// ....
ExpectedErr error 
}
```
and in test body 
```
assert.Equal(t, tc.ExpectedErr, err) 
```

Seems slightly more readable to me.

Reply:
Then I have to have in each test case the full error message including the one from the `jaeger` package in the one case e.g. 
`cannot parse jaeger env vars: cannot parse env var JAEGER_RPC_METRICS=not a boolean: strconv.ParseBool: parsing \"not a boolean\": invalid syntax`

Using `require.ErrorContains()` I can have there a substring. That is why I find it more useful.